equistore/operations/solve.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+from equistore import TensorBlock, TensorMap
+
+from . import _dispatch
+
+
+def solve(X: TensorMap, Y: TensorMap) -> TensorMap:
+    """
+    Solve a linear system among two :py:class:`TensorMap`.
+    Solve the linear equation set X * w = Y for the unknown w.
+    Where X , w, Y are all :py:class:`TensorMap`
+    """
+
+    if len(X) != len(Y) or (not np.all([key in Y.keys for key in X.keys])):
+        raise ValueError("The two input TensorMap should have the same keys")
```

Comment:
this feels like an operation we will have to do a lot, should we extract it to a separate function?